functions.photon.php

```diff
@@ -273,18 +273,18 @@ function jetpack_photon_url_scheme( $url, $scheme ) {
  * A wrapper for PHP's parse_url, prepending assumed scheme for network path
  * URLs. PHP versions 5.4.6 and earlier do not correctly parse without scheme.
  *
+ * WP ships with a wrapper for parse_url, wp_parse_url, that should be used instead.
+ *
  * @see https://php.net/manual/en/function.parse-url.php#refsect1-function.parse-url-changelog
+ * @deprecated 7.8.0 Use wp_parse_url instead.
  *
  * @param string $url The URL to parse
  * @param integer $component Retrieve specific URL component
  * @return mixed Result of parse_url
  */
 function jetpack_photon_parse_url( $url, $component = -1 ) {
-	if ( 0 === strpos( $url, '//' ) ) {
-		$url = ( is_ssl() ? 'https:' : 'http:' ) . $url;
-	}
-
-	return parse_url( $url, $component );
+	_deprecated_function( 'jetpack_photon_parse_url', 'jetpack-7.8.0', 'wp_parse_url' );
```

Comment:
Looks like you'll need to update the Photon Unit Tests as well to not use that function.

Reply:
cba4c1d9e